_posts/2023-09-20-intro-to-the-various-java-based-k8s-oc-tools.adoc

```diff
@@ -0,0 +1,607 @@
+`JAX-RS` and servlet annotations and registers the http port accordingly
+in the generated manifest file.
+
+== Eclipse JKube
+
+The Eclipse JKube can generate `Dockerfile` and do the docker image
+build, and it can generate Kubernetes/OpenShift manifests, and do the
+k8s/openshift deployment. There is an article from
+`developers.redhat.com`footnote:[https://developers.redhat.com/blog/2020/01/28/introduction-to-eclipse-jkube-java-tooling-for-kubernetes-and-red-hat-openshift[Introduction
+to Eclipse JKube: Java tooling for Kubernetes and Red Hat OpenShift |
+Red Hat Developer]] that describes the history of the project, and here
+is part of the text quoted from the article:
+
+____
+This project was not built from scratch. It’s just a refactored and
+rebranded version of
+the https://github.com/fabric8io/fabric8-maven-plugin[Fabric8 Maven
+plugin], which was a Maven plugin used in
+the http://fabric8.io/[Fabric8] ecosystem. Although the Fabric8 project
+was liked and appreciated by many people in the open source community,
+due to unfortunate reasons it could not become successful, and the idea
+of Fabric8 as an integrated development platform on top of Kubernetes
+died. Although the main project is archived, there are still active
+repositories used by the community, such as
+the https://github.com/fabric8io/docker-maven-plugin[Fabric8 Docker
+Maven
+plugin], the https://github.com/fabric8io/kubernetes-client[Fabric8
+Kubernetes client], and of course the Fabric8 Maven plugin. As
+maintainers of the Fabric8 Maven plugin, we started decoupling the
+Fabric8 ecosystem related pieces from the plugin to make a
+general-purpose Kubernetes/OpenShift plugin. We also felt there was a
+need for rebranding because most people were confused about whether this
+plugin had something to do with Fabric8. Hence, we decided to rebrand
+it, and fortunately, someone from the Eclipse foundation approached us
+to take in our project. Now, the project is being renamed to Eclipse
+JKube and can be https://github.com/eclipse/jkube[found in the Eclipse
+Foundation repos on GitHub].
+____
+
+And here is its design:
+
+____
+Eclipse JKube can be seen as a reincarnation of the Fabric8 Maven
+plugin. It contains the good parts of this plugin and offers a clean and
+smooth workflow with the tooling it provides. We refactored this plugin
+into three components: *
+https://github.com/eclipse/jkube/tree/master/jkube-kit[The JKube Kit] *
+https://github.com/eclipse/jkube/tree/master/kubernetes-maven-plugin[The
+Kubernetes Maven plugin] *
+https://github.com/eclipse/jkube/tree/master/openshift-maven-plugin[The
+OpenShift Maven plugin]
+____
+
+Unlike the Dekorate project, which focus on k8s manifest file
+generation, and provide fine-grained annotations to control the
+generation process, JKube run as a maven plugin can help to generate a
+simple `Dockerfile` based on the project used frameworks(`wildfly-jar`
+or Quarkus for
```

Comment:
This also should work with the a WildFly image generated from the wildfly-maven-plugin.

Reply:
@jamezp I have modified the text here but not sure the modification is accurate. Could you please elaborate more on this? Do you mean the `wildfly-maven-plugin` can be used to generate Dockerfile and build Docker image? If so, are there examples showing its usage? Guess I can add a section describing `wildfly-maven-plugin` usage.